mrmustard/math/backend_tensorflow.py

```diff
@@ -83,13 +90,13 @@ def astensor(self, array: Union[np.ndarray, tf.Tensor], dtype=None) -> tf.Tensor
         return tf.convert_to_tensor(array, dtype)
 
     def atleast_1d(self, array: tf.Tensor, dtype=None) -> tf.Tensor:
-        return tf.experimental.numpy.atleast_1d(self.astensor(array, dtype))
+        return tf.experimental.numpy.atleast_1d(self.cast(self.astensor(array), dtype))
```

Comment:
why this change?

Reply:
An error showed up that can be reproduced by the following snippet:

```
import tensorflow as tf

my_tensor = tf.convert_to_tensor([1, 2])
tf.convert_to_tensor(my_tensor, tf.complex128)
--> ValueError: Tensor conversion requested dtype complex128 for Tensor with dtype int32: <tf.Tensor: shape=(2,), dtype=int32, numpy=array([1, 2], dtype=int32)>
```